src/transformers/models/imagebind/modeling_imagebind.py

```diff
@@ -0,0 +1,2062 @@
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        attention_mask: torch.Tensor,
+        output_attentions: Optional[bool] = False,
+    ) -> Tuple[torch.FloatTensor]:
+        """
+        Args:
+            hidden_states (`torch.FloatTensor`): input to the layer of shape `(batch, seq_len, embed_dim)`
+            attention_mask (`torch.FloatTensor`): attention mask of size
+                `(batch, 1, tgt_len, src_len)` where padding elements are indicated by very large negative values.
+                `(config.encoder_attention_heads,)`.
+            output_attentions (`bool`, *optional*):
+                Whether or not to return the attentions tensors of all attention layers. See `attentions` under
+                returned tensors for more detail.
+        """
+        residual = hidden_states
+
+        hidden_states = self.layernorm_before(hidden_states)
+        hidden_states, attn_weights = self.self_attn(
+            hidden_states=hidden_states,
+            attention_mask=attention_mask,
+            output_attentions=output_attentions,
+        )
+        hidden_states = self.drop_path(hidden_states)
+        hidden_states = residual + hidden_states
+
+        residual = hidden_states
+        hidden_states = self.layernorm_after(hidden_states)
+        hidden_states = self.mlp(hidden_states)
+        hidden_states = self.drop_path(hidden_states)
+        hidden_states = residual + hidden_states
+
+        outputs = (hidden_states,)
+
+        if output_attentions:
+            outputs += (attn_weights,)
+
+        return outputs
+
+
+class ImageBindPostProcessor(nn.Module):
+    """
+    Post-processes ImageBind embeddings by using a normalize layer followed by an optional logit scaling layer.
+
+    Args:
+        config (Union[ImageBindTextConfig, ImageBindVisionConfig,ImageBindAudioConfig]): A configuration object that contains
+                initialization values for logit scaling.
+        dim (int, optional): The dimension along which to normalize the logits. Default is -1, which indicates the last dimension.
+        max_logit_scale (float, optional): The maximum value to which the logit scale can be clipped. Default is 100.
+    """
+
+    def __init__(
+        self,
+        config,
+        dim: int = -1,
+        max_logit_scale: float = 100,
+    ):
+        super().__init__()
+        self.dim = dim
+        self.scale_logits = config.logit_scale_init_value is not None
+
+        if self.scale_logits:
+            self.logit_scale_init = config.logit_scale_init_value
+            self.max_logit_scale = max_logit_scale
+            self.learnable = config.learnable_logit_scale
+
+            log_logit_scale = torch.ones([]) * np.log(self.logit_scale_init)
+            if self.learnable:
+                self.log_logit_scale = nn.Parameter(log_logit_scale)
+            else:
+                self.register_buffer("log_logit_scale", log_logit_scale)
```

Comment:
same comment about code paths, that is something we try to avoid. 
Is this used in released checkpoints?

Reply:
It is used [here](https://github.com/facebookresearch/ImageBind/blob/3fcf5c9039de97f6ff5528ee4a9dce903c5979b3/imagebind/models/helpers.py#L35) and then to the original modeling file.
It's also used in the released checkpoints for eg. this param: `'modality_postprocessors.audio.1.log_logit_scale'`